R/launcher-functions.R

```diff
@@ -0,0 +1,326 @@
+#' Retrieve Launcher Information
+#'
+#' Retrieve information about the launcher, as well as the different clusters
+#' that the launcher has been configured to use.
+#'
+#' @name launcher
+#' @export
+launcherGetInfo <- function() {
+  callLauncherFun("launcher.getInfo")
+}
+
+#' Check if Launcher is Available
+#'
+#' Check if the RStudio launcher is currently available.
+#'
+#' @name launcher
+#' @export
+launcherAvailable <- function() {
+  callLauncherFun("launcher.jobsFeatureAvailable")
+}
+
+#' Retrieve Job Information
+#'
+#' Retrieve information on launcher jobs.
+#'
+#' @param statuses Return only jobs whose status matches one of `statuses`.
+#'   Valid statuses are: Pending, Running, Suspended, Failed, Finished, Killed,
+#'   Canceled. When `NULL`, all jobs are returned.
+#'
+#' @param fields Return a subset of fields associated with each job object.
+#'   When `NULL`, all fields associated with a particular job are returned.
+#'
+#' @param includeSessions Boolean; include jobs which are also operating
+#'   as RStudio R sessions?
+#'
+#' @name launcher
+#' @export
+launcherGetJobs <- function(statuses = NULL,
+                            fields = NULL,
+                            includeSessions = FALSE)
+{
+  callLauncherFun("launcher.getJobs",
+                  statuses = statuses,
+                  fields = fields,
+                  includeSessions = includeSessions)
```

Comment:
As I mentioned, we'll also want to add the `tags` argument to this method, which serves as an additional optional filter along with `statuses`. Not worth holding this up for that, though. I've got a card tracking it in the RSP project.